job-server/src/spark.jobserver/JobServer.scala

```diff
@@ -26,6 +26,13 @@ import org.slf4j.LoggerFactory
 object JobServer {
   val logger = LoggerFactory.getLogger(getClass)
 
+  private def setSystemPropertiesFromConfig(config: Config) {
```

Comment:
@noorul Well, this part is easy to test, but anyway: where else should we put it? It is essential that this small piece of code is executed very early during the start-up process. As we wrote in the description for this Pull Request: we would welcome suggestions on how to avoid this copying from the config into properties altogether.